test/generator/configure_generator.cpp

```diff
@@ -0,0 +1,86 @@
+#include "Halide.h"
+
+namespace {
+
+class Configure : public Halide::Generator<Configure> {
+public:
+    GeneratorParam<int> num_extra_buffer_inputs{"num_extra_buffer_inputs", 3};
+
+    Input<Buffer<int>> input{"input", 3};
+    Input<int> bias{"bias"};
+
+    Output<Buffer<int>> output{ "output", 3 };
+
+    void configure() {
```

Comment:
In my implementation, I make configure take a `std::string` as a parameter.
This string comes from the registered name of the generator.

For example, if I have a generator class `MyGenClass` with the following specializations:
```
HALIDE_REGISTER_GENERATOR( MyGenClass, mygen_8bpp )
HALIDE_REGISTER_GENERATOR( MyGenClass, mygen_16bpp )
HALIDE_REGISTER_GENERATOR( MyGenClass, mygen_hdr )
```
the `configure()` method would receive either "mygen_8bpp", "mygen_16bpp" or "mygen_8hdr", which I can then inspect to make decisions about how to configure the inputs/outputs.

Either that, or there should be a way to query the base `Generator<>` class about the name of the registered stub that it is generating code for (I could not find an interface for that).

Reply:
Examining GeneratorParams is a better and more granular way to accomplish the same thing.

Reply:
No, because `GeneratorParam<string>` isn't an option for `GeneratorParam` :-)

No, what I'm saying is that instead of passing in a string which you have to parse to figure out the meaning(s) of, you should have GeneratorParam(s) controlling what you actually want. (In your example, an int that is 8 or 16 for instance.)

Reply:
Though it would be nice to have a generatorparam<string> for the cases where you *do* have something json-like that specifies the structure of your pipeline. Right now you'd have to get it out-of-band, e.g. by doing file io inside configure().

Reply:
Understood. Again, though, if it would it be possible for you to share an example of what your existing configure() method looks like, how it's called, etc., it would help inform the discussion, I think.

Reply:
In addition, I am OK having `configure` take no argument. I already cooked up a recipe with `GeneratorParam<>`, as you suggested, to pass the generator name.
(I am basically using `GeneratorParam<uint64_t>` to pass a `const char*` with the registered generator name to the `configure()` method.